src/api/Feed.js

```diff
@@ -184,14 +185,80 @@ export const getParsedFileActivitiesResponse = (response?: { entries: FileActivi
+                        const collaborators = {};
+
+                        if (versionsItem.action_by.length === 1) {
+                            versionsItem.uploader_display_name = versionsItem.action_by[0].name;
+                        }
+
+                        versionsItem.action_by.map(collaborator => {
+                            collaborators[collaborator.id] = { ...collaborator };
+                            return collaborator;
+                        });
+
+                        versionsItem.collaborators = collaborators;
+                    }
+                    if (versionsItem.end?.number !== undefined) {
+                        versionsItem.version_end = versionsItem.end.number;
+                        versionsItem.id = versionsItem.end.id;
+                    }
+                    if (versionsItem.start?.number !== undefined) {
+                        versionsItem.version_start = versionsItem.start.number;
+                    }
+
+                    if (versionsItem.version_start === versionsItem.version_end) {
+                        versionsItem.version_number = versionsItem.version_start;
+
+                        if (
+                            versionsItem.action_type === 'created' &&
+                            versionsItem.start?.created_at &&
+                            versionsItem.start?.created_by
+                        ) {
+                            const modifiedAt = versionsItem.start.created_at;
+                            const modifiedBy = { ...versionsItem.start.created_by };
+                            versionsItem.modified_at = modifiedAt;
+                            versionsItem.modified_by = modifiedBy;
+                        }
+                        if (
+                            versionsItem.action_type === 'trashed' &&
+                            versionsItem.start?.trashed_at &&
+                            versionsItem.start?.trashed_by
+                        ) {
+                            const trashedAt = versionsItem.start.trashed_at;
+                            const trashedBy = { ...versionsItem.start.trashed_by };
+                            versionsItem.trashed_at = trashedAt;
+                            versionsItem.trashed_by = trashedBy;
+                        }
+                        if (
+                            versionsItem.action_type === 'restored' &&
+                            versionsItem.start?.restored_at &&
+                            versionsItem.start?.restored_by
+                        ) {
+                            const restoredAt = versionsItem.start.restored_at;
+                            const restoredBy = { ...versionsItem.start.restored_by };
+                            versionsItem.restored_at = restoredAt;
+                            versionsItem.restored_by = restoredBy;
+                        }
+                    }
+
+                    return versionsItem;
```

Comment:
I was wondering if there would be a way to extract and organize some of this into separate functions to help reduce the length of the `getParsedFileActivitiesResponse` function? I'm also fine with leaving as-is if you don't think that would make sense here.

Reply:
Yeah, thought about it, there isn't really benefits to splitting though since the logic isn't reusable, and the switch statements make it more readable as well.